File: Modules/ServerTypes/OrganisationHierarchy/DepartmentForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface DepartmentForm {
    Name: StringEditor;
    Description: StringEditor;
}

export class DepartmentForm extends PrefixedContext {
    static formKey = 'OrganisationHierarchy.Department';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DepartmentForm.init)  {
            DepartmentForm.init = true;

            var w0 = StringEditor;

            initFormType(DepartmentForm, [
                'Name', w0,
                'Description', w0
            ]);
        }
    }
}

File: Modules/HumanResource/OrganisationHierarchy/Department/DepartmentForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.OrganisationHierarchy.Forms;

[FormScript("OrganisationHierarchy.Department")]
[BasedOnRow(typeof(DepartmentRow), CheckNames = true)]
public class DepartmentForm
{
    public string Name { get; set; }
    public string Description { get; set; }
}